Common/Utility/DocDbRestUtility.cs

```diff
@@ -215,31 +173,60 @@ public DocDbRestUtility(IConfigurationProvider configProvider)
         /// <param name="queryString"></param>
         /// <param name="queryParameters"></param>
         /// <returns></returns>
-        public async Task<DocDbRestQueryResult> QueryDeviceDbAsync(
+        public async Task<DocDbRestQueryResult> QueryDeviceManagementCollectionAsync(
             string queryString, Dictionary<string, Object> queryParams, int pageSize = -1, string continuationToken = null)
         {
-            WebClient client = BuildWebClient();
-            client.Headers.Set("Content-Type", "application/query+json");
-            client.Headers.Add(AUTHORIZATION_HEADER_KEY, GetAuthorizationTokenForDeviceManagementCollectionQuery("POST"));
-            client.Headers.Add("x-ms-documentdb-isquery", "true");
+            using (WebClient client = BuildWebClient())
+            {
+                string endpoint = string.Format("{0}dbs/{1}/colls/{2}/docs", _docDbEndpoint, _dbId, _collectionId);
+
+                var result = new DocDbRestQueryResult();
+
+                string response = await QueryDocDbInternal(client, endpoint, queryString, queryParams, DOCUMENTS_RESOURCE_TYPE, _collectionId, pageSize, continuationToken);
+                JObject responseJobj = JObject.Parse(response);
+                JToken documents = responseJobj.GetValue("Documents");
+                if (documents != null)
+                {
+                    result.Documents = (JArray)documents;
+                }
+
+                WebHeaderCollection responseHeaders = client.ResponseHeaders;
+
+                string count = responseHeaders["x-ms-item-count"];
+                if (!string.IsNullOrEmpty(count))
+                {
+                    result.TotalDocuments = int.Parse(count);
+                }
+                result.ContinuationToken = responseHeaders["x-ms-continuation"];
+
+                return result;
+            }
+        }
+
+        private async Task<string> QueryDocDbInternal(WebClient preparedWebClient, string endpoint, string queryString, Dictionary<string, Object> queryParams, 
+            string resourceType, string resourceId, int pageSize = -1, string continuationToken = null)
+        {
+            preparedWebClient.Headers.Set("Content-Type", "application/query+json");
```

Comment:
Actually why are you passing in WebClient...  I don't see anything being "prepared" just a using wrapper around this function.  Its cleaner if this function creates and disposes of its own client.